### PR TITLE
Immune deconv

### PR DESCRIPTION
**Pull request by komalsrathi**
### Purpose/implementation Section

Immune deconvolution

#### What scientific question is your analysis addressing?

Some plot changes per: https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/519

#### What was your approach?

1. Changed colors for correlation plot
2. Separated heatmaps of brain/non-brain tumors.
3. Converted png format to pdf so that they can be edited in Illustrator or other editors if needed.

#### What GitHub issue does your pull request address?

https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/519

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes

#### What types of results are included (e.g., table, figure)?

PDF files

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
Hi @komalsrathi, thanks for this add! It looks like this is failing continuous integration with the following error:

```
>>> Running mcp_counter
[1] "Writing output to file.."
[1] "Done!"
Deconvolution finished...
Create summary plots
[1] "Overall Pearson Correlation:  0.67"
Warning messages:
1: In cor(xCell, MCPcounter) : the standard deviation is zero
2: In cor(xCell, MCPcounter) : the standard deviation is zero
null device 
          1 
Error: Unknown columns `Lymphomas (1)` and `Histiocytic tumor (5)` 
Backtrace:
     █
  1. └─global::create.heatmap(...)
  2.   └─`%>%`(...)
  3.     ├─base::withVisible(eval(quote(`_fseq`(`_lhs`)), env, env))
  4.     └─base::eval(quote(`_fseq`(`_lhs`)), env, env)
  5.       └─base::eval(quote(`_fseq`(`_lhs`)), env, env)
  6.         └─`_fseq`(`_lhs`)
  7.           └─magrittr::freduce(value, `_function_list`)
  8.             └─function_list[[i]](value)
  9.               ├─dplyr::select(., non.brain.tumors)
 10.               └─dplyr:::select.data.frame(., non.brain.tumors)
 11.                 └─tidyselect::vars_select(tbl_vars(.data), !!!enquos(...))
 12.                   └─tidyselect:::vars_select_eval(.vars, quos)
 13.                     └─purrr::map_if(ind_list, is_character, match_st
Execution halted

Exited with code exit status 1
```

It seems this is due to the hardcoding of the non-brain tumors here: https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/536/files#diff-eb64c2b951d2fd18ea2f67c65de779edR69

I assume that the numbers in parentheses are the _sample size_ for a particular histology. It's quite possible that there are no Lymphoma or Histiocytic tumor in [the subset files that we use for CI](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/README.md#continuous-integration-ci) and if there are the sample size is likely different.

One idea about how to make this more robust off the top of my head:

* Make a vector of brain cancers and split up the deconvolution matrix (if I'm following correctly) into brain cancer and non-brain cancer via regex, where you use negation to get your non-brain cancer matrix. Then you can check to make sure that the non-brain cancer matrix is not empty before proceeding. My rationale for the brain cancer vector + regex approach is that there will always be some samples that meet the criterion for brain cancer in the files for continuous integration.

**Comment by komalsrathi** [comment]
@jaclyn-taroni I have made some changes. If this fails, I will try some other approach. Will let you know once ready to review. Thank you!

**Comment by komalsrathi** [comment]
@jaclyn-taroni I think the immune deconv scripts worked but CI is failing at the Tumor mutation burden step.

**Comment by jaclyn-taroni** [comment]
Okay, thanks! Let me see if updating the branch fixes it.

File: analyses/immune-deconv/02-summary-plots.R [hunk 1]
```diff
@@ -64,13 +67,39 @@ create.heatmap <- function(deconv.method, title) {
...
-  
-  title <- paste0(title,"\nAverage immune scores normalized by rows")
-  pheatmap(mat = t(deconv.method), fontsize = 10, 
-           scale = "column", angle_col = 45,
-           main = title, annotation_legend = T, cellwidth = 15, cellheight = 15)
+  # plot non-brain and brain tumors separately
+  pdf(file = fileout, width = 13, height = 8)
+  # non-brain tumors
+  mat <- deconv.method %>% 
+    select(grep(paste0(non.brain.tumors, collapse="|"), colnames(deconv.method), value = TRUE))
```
Comment (jaclyn-taroni) [inline analyses/immune-deconv/02-summary-plots.R]: I believe you could use [`tidyselect::matches`](https://www.rdocumentation.org/packages/tidyselect/versions/1.0.0/topics/select_helpers) here: 
```suggestion
    select(matches(paste0(non.brain.tumors, collapse="|"))) 
```
(untested)

**Pull request merged**